ili9341_stm32_parallel8.h

```diff
@@ -87,87 +87,221 @@ SOFTWARE.
 #define ILI_PWCTR6  0xFC
 */
 
+#ifdef USER_DEFAULT_PLATFORM
+    // Color definitions
+    #define ILI_COLOR_BLACK       0x0000  ///<   0,   0,   0
+    #define ILI_COLOR_NAVY        0x000F  ///<   0,   0, 123
+    #define ILI_COLOR_DARKGREEN   0x03E0  ///<   0, 125,   0
+    #define ILI_COLOR_DARKCYAN    0x03EF  ///<   0, 125, 123
+    #define ILI_COLOR_MAROON      0x7800  ///< 123,   0,   0
+    #define ILI_COLOR_PURPLE      0x780F  ///< 123,   0, 123
+    #define ILI_COLOR_OLIVE       0x7BE0  ///< 123, 125,   0
+    #define ILI_COLOR_LIGHTGREY   0xC618  ///< 198, 195, 198
+    #define ILI_COLOR_DARKGREY    0x7BEF  ///< 123, 125, 123
+    #define ILI_COLOR_BLUE        0x001F  ///<   0,   0, 255
+    #define ILI_COLOR_GREEN       0x07E0  ///<   0, 255,   0
+    #define ILI_COLOR_CYAN        0x07FF  ///<   0, 255, 255
+    #define ILI_COLOR_RED         0xF800  ///< 255,   0,   0
+    #define ILI_COLOR_MAGENTA     0xF81F  ///< 255,   0, 255
+    #define ILI_COLOR_YELLOW      0xFFE0  ///< 255, 255,   0
+    #define ILI_COLOR_WHITE       0xFFFF  ///< 255, 255, 255
+    #define ILI_COLOR_ORANGE      0xFD20  ///< 255, 165,   0
+    #define ILI_COLOR_GREENYELLOW 0xAFE5  ///< 173, 255,  41
+    #define ILI_COLOR_PINK        0xFC18  ///< 255, 130, 198
+
+#elif DSO138_PLATFORM
+    #define	RB 0  // Red bits position
+    #define	GB 5  // Green bits position
+    #define	BB 11 // Blue bits position
+    #define	RGB(R,G,B) (((R >> 3) << RB) | ((G >> 2) << GB) | ((B >> 3) << BB))
```

Comment:
This macro is placing Blue component in the MSB (position 11)  and red in the LSB (position 0). As a result, it's creating BGR color instead of RGB. 
To create RGB, just swap the bit position of Red and Blue. Change your code to:

```C
#define	R_POS			11	// Red last bit position
#define	G_POS			5 	// Green last bit position
#define	B_POS			0	// Blue last bit position
#define	RGB(R,G,B)		(((uint16_t)(R >> 3) << R_POS) | ((uint16_t)(G >> 2) << G_POS) | ((uint16_t)(B >> 3) << B_POS))
```
The compiler will perform typecasting automatically, but it's safer to do it explicitly.

Reply:
That's a valid point. Originally I've tested color constants from your code and found that in DSO138 display colors bits are "swapped" - e.g. line with `ILI_COLOR_RED` becomes blue on a screen and vice versa.

Reply:
If DSO138's display wants color in BGR format, keep that macro but rename it to BGR. Also, add the RGB macro I suggested so that users have both options.

Reply:
Ok